src/modules/map/components/olMap/handler/measure/OlMeasurementHandler.js

```diff
@@ -281,32 +284,59 @@ export class OlMeasurementHandler extends OlLayerHandler {
 
 	_register(store) {
 		return [
+			observe(store, state => state.measurement.finish, () => this._finish()),
 			observe(store, state => state.measurement.reset, () => this._startNew()),
-			observe(store, state => state.measurement.remove, () => this._removeSelectedFeatures())];
+			observe(store, state => state.measurement.remove, () => this._remove())];
 	}
 
 
 	_removeLast(event) {
+		if ((event.which === 46 || event.keyCode === 46) && !/^(input|textarea)$/i.test(event.target.nodeName)) {
+			this._remove();
+		}
+	}
+
+	_remove() {
 		if (this._draw && this._draw.getActive()) {
-			if ((event.which === 46 || event.keyCode === 46) && !/^(input|textarea)$/i.test(event.target.nodeName)) {
-				this._draw.removeLastPoint();
-				if (this._pointCount === 2) {
-					this._reset();
-				}
-				if (this._lastPointerMoveEvent) {
-					this._draw.handleEvent(this._lastPointerMoveEvent);
-				}
+			
+			this._draw.removeLastPoint();
+			if (this._pointCount === 1) {
+				this._startNew();
+			}
+			if (this._lastPointerMoveEvent) {
+				this._draw.handleEvent(this._lastPointerMoveEvent);
 			}
 		}
 
 		if (this._modify && this._modify.getActive()) {
-			if ((event.which === 46 || event.keyCode === 46) && !/^(input|textarea)$/i.test(event.target.nodeName)) {
-				this._removeSelectedFeatures();
+		
+			this._removeSelectedFeatures();
+		}
+	}
+
+	_finish( ) {
+		if (this._draw.getActive()) {
+			if (this._activeSketch) {
+				this._draw.finishDrawing();
+				const event = new Event('click');
+				event.clientX = this._map.getView().getCenter()[0];
+				event.clientY = this._map.getView().getCenter()[1];
+				event.pageX = this._map.getView().getCenter()[0];
+				event.pageY = this._map.getView().getCenter()[1];
+				event.shiftKey = false;
+				const mapEvent = new MapBrowserEvent('click', this._map, event);
+				this._map.dispatchEvent(mapEvent);		
```

Comment:
This hack will be refactored out....probably to a util-class or a better solution comes in... :|

Reply:
I think we can live with that for now, just a comment "why this is so" would be fine...